File: TemplateMongo/TemplateMongo/Model/Vehicle.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.Collections.Generic;

namespace TemplateMongo.Model
{
    public class Vehicle
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        [BsonElement("plateNum")]
        public string plateNum { get; set; }

        public string manufacture { get; set; }
       
        public string model { get; set; }

        public int km { get; set; }
        public string tireSize { get; set; }
        //public Dictionary<int,Tire> carTires { get; set; }
        public List<Tire> tires { get; set; }
        public int numOfTires { get; set; }
        public Vehicle()
        {
            this.tires = new List<Tire>();
            this.numOfTires = 5;
            for (int i = 0; i < this.numOfTires; i++)
            {
                Tire tire = new Tire();
                tire.location = i + 1;
                this.tires.Add(tire);
            }
        }
    }
}

File: client/src/app/model/vehicle.ts
import { Tire } from './tire';

export class Vehicle {
  public Id:string;
public plateNum:string;
public manufacture:string;
public model:string;
public km:number = 1;
public tireSize:string;
// public carTires: { [key: number]: Tire; } = {};
//public carTires: Map<number,Tire> = new Map<number,Tire>();
public tires: Array<Tire> = new Array<Tire>();
public numOfTires:number;

/**
 *
 */
constructor() {
  this.plateNum ="";
  this.numOfTires = 5;
  for(let i=0;i<this.numOfTires;i++){
    let tire = new Tire();
    tire.location = i+1;
    this.tires.push(tire);
    //this.carTires.set(i+1,tire);
  }
  //
}

}
